File: RiotShopBackEnd/RiotShopBackEnd/DTOs/UserResponse.cs
namespace RiotShopBackEnd.DTOs;

public class UserResponse
{
    public int UserId { get; set; }
    public string Username { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? FullName { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Address { get; set; }
    public bool IsAdmin { get; set; }
    public decimal Balance { get; set; }
    public DateTime CreatedAt { get; set; }
    public string? AvatarUrl { get; set; }
    public DateTime? LastLogin { get; set; }
}

File: riotshop/riotshop/app/src/main/java/com/example/riotshop/models/UserResponse.java
package com.example.riotshop.models;

import com.google.gson.annotations.SerializedName;

public class UserResponse {
    @SerializedName("userId")
    private int userId;
    
    @SerializedName("username")
    private String username;
    
    @SerializedName("email")
    private String email;
    
    @SerializedName("fullName")
    private String fullName;
    
    @SerializedName("phoneNumber")
    private String phoneNumber;
    
    @SerializedName("address")
    private String address;
    
    @SerializedName("isAdmin")
    private boolean isAdmin;
    
    @SerializedName("balance")
    private double balance;
    
    @SerializedName("createdAt")
    private String createdAt;
    
    @SerializedName("avatarUrl")
    private String avatarUrl;
    
    @SerializedName("lastLogin")
    private String lastLogin;

    // Getters
    public int getUserId() {
        return userId;
    }

    public String getUsername() {
        return username;
    }

    public String getEmail() {
        return email;
    }

    public String getFullName() {
        return fullName;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public String getAddress() {
        return address;
    }

    public boolean isAdmin() {
        return isAdmin;
    }

    public double getBalance() {
        return balance;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public String getAvatarUrl() {
        return avatarUrl;
    }

    public String getLastLogin() {
        return lastLogin;
    }
}

